core/src/main/java/com/linecorp/armeria/server/http/healthcheck/SettableHttpHealthCheckService.java

```diff
@@ -0,0 +1,101 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.healthcheck;
+
+import java.util.Optional;
+
+import com.linecorp.armeria.common.http.AggregatedHttpMessage;
+import com.linecorp.armeria.common.http.HttpData;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponseWriter;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * This is more rich implementation of {@link HttpHealthCheckService}.
+ * tl;dr
+ *
+ * <h2>Example as default</h2>
+ * <pre>{@code
+ * Server server = new ServerBuilder()
+ *         .defaultVirtualHost(new VirtualHostBuilder()
+ *                 .serviceUnder("/health", new SettableHttpHealthCheckService())
+ *                 .build())
+ *         .build();
+ * }</pre>
+ * Default config uses path ../on and ../off
+ * You can also use your own path matching.
+ *
+ * <pre>{@code
+ * ...
+ *                 .serviceUnder("health", new SettableHttpHealthCheckService() {
+ *                         public Optional<Boolean> mode(HttpRequest req) {
+ *                                 return Optional.empty();
+ *                         }
+ *                 })
+ *                 .build())
+ * ...
+ * }</pre>
+ */
+public class SettableHttpHealthCheckService extends HttpHealthCheckService {
+
+    private static final AggregatedHttpMessage TURN_ON_RES = AggregatedHttpMessage
+            .of(HttpStatus.OK, HttpData.ofAscii("turn on"));
+    private static final AggregatedHttpMessage TURN_OFF_RES = AggregatedHttpMessage
+            .of(HttpStatus.OK, HttpData.ofAscii("turn off"));
+
+    @Override
+    protected void doPost(ServiceRequestContext ctx, HttpRequest req, HttpResponseWriter res) throws Exception {
+        res.respond(updateHealthStatus(req));
+    }
+
+    /**
+     * Updates health status using request data.
+     * @param req HttpRequest
+     */
+    private AggregatedHttpMessage updateHealthStatus(HttpRequest req) {
+        Optional<Boolean> mode = mode(req);
+
+        if (!mode.isPresent()) {
+            return AggregatedHttpMessage.of(HttpStatus.BAD_REQUEST);
+        }
+
+        boolean tobe = mode.get();
+
+        serverHealth.setHealthy(tobe);
+
+        return tobe ? TURN_ON_RES : TURN_OFF_RES;
+    }
+
+    /**
+     * Judge the turning mode.
+     * True means turn on, False is turn off.
+     * If not present, Not supported.
+     * @param req HttpRequest
+     */
+    protected Optional<Boolean> mode(HttpRequest req) {
```

Comment:
- A user cannot override this method to implement the case where a user wants the command to be put in the HTTP content. `req` could be `AggregatedHttpMessage` for example.
- As we discussed yesterday evening, could you make the default implementation use HTTP content rather than the request path? Also, please add the test case that tests the case where a user overrides this method to implement the alternative logic?